akka-cluster-sharding/src/test/scala/akka/cluster/sharding/LeastShardAllocationStrategySpec.scala

```diff
@@ -129,6 +129,119 @@ class LeastShardAllocationStrategySpec extends AkkaSpec {
         Set("003", "004"))
     }
 
+    "rebalance from region with most number of shards [3, 0, 0], rebalanceThreshold=1, rebalanceNumber=2" in {
+      val allocationStrategy = new LeastShardAllocationStrategy(
+        rebalanceThreshold = 1,
+        maxSimultaneousRebalance = 10,
+        rebalanceNumber = 2,
+        rebalanceFactor = 0.0)
+      val allocations = createAllocations(aCount = 3)
+
+      allocationStrategy.rebalance(allocations, Set.empty).futureValue should ===(Set("001"))
```

Comment:
should this not be 2 rebalanced the first time as we set rebalanceNumber to 2?

Perhaps we'd get better results here if we rounded up when calculating `evenBalance`

Reply:
I think that would be a very different rebalancing strategy to what we have now. What we have now, we consider the region with the most shards, and the region with the least shards, and we do a limited amount of work to get them even. If the total number of shards between those two regions is odd, then at best, the region with the most shards would (and should) end up with one more shards then the region with the least (otherwise, in a situation where the rest of the regions have a balanced number of shards, you're potentially handing off a shard only for it to be allocated back to the same shard region).

So unless we significantly change rebalancing to take into consideration multiple regions at once, I think this is the correct behaviour.

Reply:
And maybe that would be a good rebalancing strategy, take the total number of shards, ceiling divide by the total number of regions, and then attempt to evenly take shards from regions with more than that number up to the configured limits, favouring taking shards from the regions with the most. But it should probably be a new allocation strategy, I don't think it could be implemented in a way such that the behaviour is the same as existing configurations.